Add initial support for ZWave JS (#1766)
home-assistant/addons: zwave_js/DOCS.md

> # Home Assistant Add-on: Zwave JS

Reviewer: ```suggestion
# Home Assistant Add-on: Z-Wave JS
```

## Installation

Follow these steps to get the add-on installed on your system:

1. Navigate in your Home Assistant frontend to **Supervisor** -> **Add-on Store**.

> 2. Find the "Zwave JS" add-on and click it.

Reviewer: ```suggestion
2. Find the "Z-Wave JS" add-on and click it.
```

3. Click on the "INSTALL" button.

> 4. This add-on currently requires to have the Mosquitto add-on installed.
>    Please make sure to install and set up that add-on before continuing.

Reviewer: ```suggestion
```

## How to use

The add-on needs to know where your Z-Wave stick can be found, and therefore,
you'll need to configure the add-on to point to the right device.

If you're using Home Assistant you may find the correct value for this on the
`Supervisor -> System -> Host system -> Hardware` page. It is recommended
to use a "by-id" path to the device if one exists, as it is not subject to
change if other devices are added to the system.

1. Replace `null` in the `device` option in the add-on configuration and specify
   the device name in quotes: e.g., something like
   `"/dev/serial/by-id/usb-0658_0200-if00"`,
   `"/dev/ttyUSB0"`, `"/dev/ttyAMA0"`, or `"/dev/ttyACM0"`.
2. Set your 16-byte network key in the form `0x01, 0x02...` used in order to
   connect securely to compatible devices. It is recommended that a network key
   is configured as security enabled devices may not function correctly if they
   are not added securely.
3. Click on "SAVE" to save the add-on configuration.
4. Start the add-on.

> 5. Add the OpenZWave integration to Home Assistant, see documentation:
>    <https://www.home-assistant.io/integrations/ozw>

Reviewer 2: ```suggestion
5. Add the Z-Wave JS integration to Home Assistant, see documentation:
   <https://www.home-assistant.io/integrations/zwave_js>
```

> After installing and starting this add-on, access the ozw-admin interface using
> the "OPEN WEBUI" button.

Reviewer 2: ```suggestion
```
Not valid for zwave-js-server

In most cases this looks like one of the following:

- `"/dev/serial/by-id/usb-0658_0200-if00"`
- `"/dev/ttyUSB0"`
- `"/dev/ttyAMA0"`
- `"/dev/ttyACM0"`

### Option `network_key`

Security Z-Wave devices require a network key before being added to the network.
You must set the `network_key` configuration option to use a network key before
adding these devices.

To generate a network key, you can use the following script in, e.g., the SSH
add-on:

```bash
cat /dev/urandom | tr -dc '0-9A-F' | fold -w 32 | head -n 1 | sed -e 's/\(..\)/0x\1, /g' -e 's/, $//'
```

You can also use sites like this one to generate the required data,
just remember to put `0x` before each pair of characters:

<https://www.random.org/cgi-bin/randbyte?nbytes=16&format=h>

Ensure you keep a backup of this key. If you have to rebuild your system and
don't have a backup of this key, you won't be able to reconnect to any security
devices. This may mean you have to do a factory reset on those devices, and your
controller, before rebuilding your Z-Wave network.

### Option `emulate_hardware` (optional)

If you don't have a USB stick, you can use a fake stick.

## Known issues and limitations

> - Your hardware needs to be compatible with Zwave JS library

Reviewer: ```suggestion
- Your hardware needs to be compatible with Z-Wave JS library
```

## Support

Got questions?

You have several options to get them answered:

- The [Home Assistant Discord Chat Server][discord].
- The Home Assistant [Community Forum][forum].
- Join the [Reddit subreddit][reddit] in [/r/homeassistant][reddit]

In case you've found a bug, please [open an issue on our GitHub][issue].

[discord]: https://discord.gg/c5DvZ4e
[forum]: https://community.home-assistant.io
[issue]: https://github.com/home-assistant/hassio-addons/issues
[reddit]: https://reddit.com/r/homeassistant

> [vnc-service-url]: vnc://homeassistant.local:5900
> [vnc-viewer]: https://bintray.com/tigervnc/stable/tigervnc/

Reviewer: ```suggestion
```
